### PR TITLE
Refresh template messages for 360dialog Channels

### PR DESCRIPTION
**Pull request by jcbalmeida**
* Creates a scheduled task to refresh the message tempaltes through 360dialog API (https://docs.360dialog.com/api/whatsapp-api/message-templates-api)
* The task is registered within settings.CELERYBEAT_SCHEDULE (15 min. interval, like wpp refresh task)
* Adds the relative haml templates (templates.haml and sync_logs.haml)

There is some differences between the official WhatsApp API and 360dialog, an important one is the templates list response does not provide the templates ids. I managed it using the key "language/name" as external_id.

### CODE REVIEW

File: temba/channels/types/dialog360/type.py [hunk 1]
```diff
@@ -1,21 +1,101 @@
...
+from temba.templates.models import TemplateTranslation
 
 from ...models import ChannelType
 
+STATUS_MAPPING = dict(
+    submitted=TemplateTranslation.STATUS_PENDING,
+    approved=TemplateTranslation.STATUS_APPROVED,
+    rejected=TemplateTranslation.STATUS_REJECTED,
+)
+LANGUAGE_MAPPING = dict(
```
Comment (rowanseymour) [inline temba/channels/types/dialog360/type.py:20]: This seems to be a duplication of what's in `whatsapp/type.py`. It also seems we could figure this stuff out programatically no? There are other places where we lean on pycountry to map between language codes.

File: templates/channels/types/dialog360/sync_logs.haml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+-block subtitle
+  -trans "360dialog Templates Events"
+
+-block content
+  -if sync_logs
+
+    %table.list.table.table-condensed
+      %thead
+        %tr
+          %th{ style:'width:150px' }
```
Comment (rowanseymour) [inline templates/channels/types/dialog360/sync_logs.haml:20]: use new HAML syntax - `%th(style="width:150px")`. I know there's there's lots of code using the older syntax but we're trying to use the new syntax for new code

File: templates/channels/types/dialog360/sync_logs.haml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+-load smartmin temba compress humanize channels
+-load i18n tz
+
+-block fields
+
+-block title
+  {{ object.name }}
+
+-block subtitle
+  -trans "360dialog Templates Events"
```
Comment (rowanseymour) [inline templates/channels/types/dialog360/sync_logs.haml:12]: `360dialog Template Events` maybe?

**Review by rowanseymour — CHANGES_REQUESTED** [review]
Probably can't merge this this week as we're still in code-freeze for 6.0. 

Does seem to be some duplication with this and the existing whatsapp channel around template management.. wondering how we can reduce that..

**Comment by matmsa27** [comment]
Hi @rowanseymour , can we move with this evolution on WhatsApp 360 Channel? This feature is very special because can improve the use of this channel by automatically way to getting template message from 360 base, and use on Rapidpro and then sending to the contacts.

File: temba/channels/types/dialog360/tasks.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+from .type import STATUS_MAPPING
+
+logger = logging.getLogger(__name__)
+
+VARIABLE_RE = re.compile(r"{{(\d+)}}")
+LANGUAGE_RE = re.compile(r"([a-zA-Z]{2,3})_?([a-zA-Z]{2})?")
+
+
+def _calculate_variable_count(content):
```
Comment (rowanseymour) [inline temba/channels/types/dialog360/tasks.py:25]: Still have quite a bit of functionality like this duplicated between `types/dialog360` and `types/whatsapp`, and perhaps soon `types/twilio_whatsapp` and `types/textit_whatsapp`. Maybe this stuff could move to `temba.utils.whatsapp`? @nicpottier @norkans7 ?

File: templates/channels/types/dialog360/sync_logs.haml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+-extends "channels/channel_read.html"
```
Comment (rowanseymour) [inline templates/channels/types/dialog360/sync_logs.haml:1]: This is also duplicates `templates/channels/types/whatsapp/sign_logs.haml`. Seems like we could have a shared Sync Logs view. Maybe not something we need to address now tho.

**Review by nicpottier — CHANGES_REQUESTED** [review]
Agree with Rowan that some effort should be taken to share functionality with the main WhatsApp channel as most of this functionality is identical. Can you refactor to make sure that's the case?

**Comment by matmsa27** [comment]
I agree with you guys, we can refactor this. I believe the task for refreshing templates can be on `temba.utils.whatsapp` and all channels (WhatsApp, 360, Kaleyra, TextIt and Twilio) for WhatsApp can use the same task, only changing some parameters like URL, headers, and other points that we can discuss, what do you think?

**Comment by rowanseymour** [comment]
I think my preference would be actually be `temba.channels.whatsapp` if this is going to include tasks and UI stuff

**Comment by rowanseymour** [comment]
Closed in favor of https://github.com/nyaruka/rapidpro/pull/3301

**Pull request closed without merging**